Change default list item label to first field of class

witalosk opened the pull request:
# Changes
- Changed default list item label to first field of class if the field is string
![image](https://github.com/user-attachments/assets/3d25e27d-c5e4-4fd8-a73f-dc3b8e63b8ab)

fuqunaga commented:
Thanks for the PR!
I agree with the direction to align with the inspector's behavior.

## Keep `Item {index}`

I suggest displaying `Item {index}` in the following cases:

1. When the string is null or empty.
2. When the `NonSerializedAttribute` is applied.
3. When the UI is unregistered using `UICustom.UnregisterPropertyOrFields<SimpleClass>("stringValue");`

Cases 1 and 2 match the inspector's behavior.
For case 2 and 3, consider using `TypeUtility.GetUITargetFieldNames()` or `PropertyOrFieldBinder.Create()` for consistency, referencing `BinderToElement`.

https://github.com/fuqunaga/RosettaUI/blob/872680fc2ea862ba430f6343204e6b343bc09e3d/Packages/ga.fuquna.rosettaui/Runtime/UI/BinderToElement/BinderToElement_Field.cs#L129-L131


## No Action Required: Ignore Attribute Customization

Even if custom attributes hide the UI of the first string, the inspector still displays the string value as the label.
(RosettaUI will add this feature in #38)
Therefore, we can ignore attribute customization when determining the label.


```
public class TestStringAttribute : PropertyAttribute
{
}

[CustomPropertyDrawer(typeof(TestStringAttribute))]
public class TestStringDrawer : PropertyDrawer
{
    public override void OnGUI(Rect position, SerializedProperty property, GUIContent label)
    {
    }
}

public class SimpleClass
{
    [TestString]
    public string stringValue;
    public float floatValue;
    private int _privateValue; // will be ignored
}
```
![image](https://github.com/user-attachments/assets/63c5a281-10f2-4818-8dba-d9641d879596)

witalosk commented:
@fuqunaga 
I have fixed the points you raised!
And I changed it to use MetadataToken to get the first field of the class.
(It seems to be speculated by the forum that Unity also uses this method to determine the order of the fields in the class.)

### References
- https://discussions.unity.com/t/retrieving-order-of-declaration-of-class-members/881963/7
- https://stackoverflow.com/questions/8067493/if-getfields-doesnt-guarantee-order-how-does-layoutkind-sequential-work

fuqunaga commented:
Thanks for the great work!
